# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../cuneiform/a-html.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

=== modified file 'cuneiform_src/Kern/rout/src/text.cpp'
--- ./cuneiform_src/Kern/rout/src/text.cpp	2008-07-29 09:38:01 +0000
+++ ./cuneiform_src/Kern/rout/src/text.cpp	2008-09-08 14:01:28 +0000
@@ -244,15 +244,27 @@
 
 		// ”гловые скобки в HTML замен€ютс€ на круглые
 		case '<':
-			if (gFormat == ROUT_FMT_HTML)
-				c2 = '(';
-			break;
-
 		case '>':
-			if (gFormat == ROUT_FMT_HTML)
-				c2 = ')';
+			if (gFormat == ROUT_FMT_HTML) {
+				*gMemCur++ = '&';
+				*gMemCur++ = (c1 == '<' ? 'l' : 'g');
+				*gMemCur++ = 't';
+				*gMemCur++ = ';';
+				return TRUE;
+			}
 			break;
 
+		case '&':
+                        if(gFormat == ROUT_FMT_HTML) {
+                            *gMemCur++ = '&';
+                            *gMemCur++ = 'a';
+                            *gMemCur++ = 'm';
+                            *gMemCur++ = 'p';
+                            *gMemCur++ = ';';
+                            return TRUE;
+                        }
+                        break;
+
 		// ƒифтонг oe / OE
 		case oe_deaf_sound:
 			if( FALSE == (

